pyensign/stream.py

```diff
@@ -175,6 +175,13 @@ async def connect(self):
         # Save the topics from the ready message
         self._parse_topics(rep.ready.topics)
 
+        # Set the topic ID for publish requests
+        try:
+            self._update_topic()
+        except EnsignTopicNotFoundError as e:
+            await self.close()
+            raise e
```

Comment:
Bit of a chicken and the egg, the client doesn't know what topics are available until opening the stream, so my solution is to close the publisher if the topic is not in the map returned by the server. Exceptions here should be caught and returned to the user as an `UnknownTopicError`.